src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -884,6 +937,187 @@ internal class OnboardingSupervisor(
         }
     }
 
+    private fun retrieveSkipWithdrawalRouteExchange(
+        state: PerpetualState?,
+        decimals: Int,
+        gas: BigDecimal,
+        accountAddress: String,
+        sourceAddress: String,
+        subaccountNumber: Int?,
+    ) {
+        val toChain = helper.configs.nobleChainId()
+        val toToken = helper.configs.nobleDenom()
+        val toAddress = state?.input?.transfer?.address
+        val usdcSize = helper.parser.asDecimal(state?.input?.transfer?.size?.usdcSize)
+        val fromAmount = if (usdcSize != null && usdcSize > gas) {
+            ((usdcSize - gas) * Numeric.decimal.TEN.pow(decimals)).toBigInteger()
+        } else {
+            null
+        }
+        val fromChain = helper.environment.dydxChainId
+        val nativeChainUSDCDenom = helper.environment.tokens["usdc"]?.denom
+        val fromAmountString = helper.parser.asString(fromAmount)
+        val url = helper.configs.skipV2MsgsDirect()
+        val fromAddress = accountAddress
+        if (
+            fromAmount != null &&
+            fromAmount > 0 &&
+            fromAmountString != null &&
+            fromChain != null &&
+            toToken != null &&
+            toChain != null &&
+            nativeChainUSDCDenom != null
+        ) {
+            val body: Map<String, Any> = mapOf(
+                "amount_in" to fromAmountString,
+                "source_asset_denom" to nativeChainUSDCDenom,
+                "source_asset_chain_id" to fromChain,
+                "dest_asset_denom" to toToken,
+                "dest_asset_chain_id" to toChain,
+                "chain_ids_to_addresses" to mapOf(
+                    fromChain to fromAddress,
+                    toChain to toAddress,
+                ),
+                "slippage_tolerance_percent" to SLIPPAGE_PERCENT,
+            )
+
+            val oldState = stateMachine.state
+            val header = iMapOf(
+                "Content-Type" to "application/json",
+            )
+            helper.post(url, header, body.toJsonPrettyPrint()) { _, response, code, headers ->
+                if (response != null) {
+                    update(stateMachine.squidRoute(response, subaccountNumber ?: 0, null), oldState)
+                } else {
+                    Logger.e { "retrieveSkipWithdrawalRouteExchange error, code: $code" }
+                }
+            }
+        }
+    }
+
+    private fun retrieveSkipWithdrawalRouteNonCCTP(
+        state: PerpetualState?,
+        decimals: Int,
+        gas: BigDecimal,
+        accountAddress: String,
+        sourceAddress: String,
+        subaccountNumber: Int?,
+    ) {
+        val toChain = state?.input?.transfer?.chain
+        val toToken = state?.input?.transfer?.token
+        val toAddress = state?.input?.transfer?.address
+        val usdcSize = helper.parser.asDecimal(state?.input?.transfer?.size?.usdcSize)
```

Comment:
for any of these that are required, can we just return early?
`val toChain = state?.input?.transfer?.chain ?: return`

same for other methods in this pr